docker/kubernetes-tentacle/scripts/configure-tentacle.sh

```diff
@@ -0,0 +1,211 @@
+            exit 1
+        fi
+    fi
+
+    if [[ -z "$ServerUrl" ]]; then
+        echo "Please specify an Octopus Server with the 'ServerUrl' environment variable" >&2
+        exit 1
+    fi
+
+    if [[ -z "$TargetEnvironment" ]]; then
+        echo "Please specify an environment name with the 'TargetEnvironment' environment variable" >&2
+        exit 1
+    fi
+
+    if [[ -z "$TargetRole" ]]; then
+        echo "Please specify a role name with the 'TargetRole' environment variable" >&2
+        exit 1
+    fi
+
+    echo " - server endpoint '$ServerUrl'"
+    echo " - api key '##########'"
+
+    if [[ ! -z "$ServerCommsAddress" || ! -z "$ServerPort" ]]; then
+        echo " - communication mode 'Kubernetes' (Polling)"
+
+        if [[ ! -z "$ServerCommsAddress" ]]; then
+            echo " - server comms address $ServerCommsAddress"
+        fi
+        if [[ ! -z "$ServerPort" ]]; then
+            echo " - server port $ServerPort"
+        fi
+    else
+        echo " - communication mode 'Kubernetes' (Listening)"
+        echo " - registered port $ListeningPort"
+    fi
+
+    echo " - environment '$TargetEnvironment'"
+    echo " - role '$TargetRole'"
+    echo " - host '$PublicHostNameConfiguration'"
+
+    if [[ ! -z "$TargetName" ]]; then
+        echo " - name '$TargetName'"
+    fi
+    if [[ ! -z "$TargetTenant" ]]; then
+        echo " - tenant '$TargetTenant'"
+    fi
+    if [[ ! -z "$TargetTenantTag" ]]; then
+        echo " - tenant tag '$TargetTenantTag'"
+    fi
+    if [[ ! -z "$TargetTenantedDeploymentParticipation" ]]; then
+        echo " - tenanted deployment participation '$TargetTenantedDeploymentParticipation'"
+    fi
+    if [[ ! -z "$Space" ]]; then
+        echo " - space '$Space'"
+    fi
+}
+
+function configureTentacle() {
+    tentacle create-instance --instance "$instanceName" --config "$configurationDirectory/tentacle.config"
+
+    echo "Setting directory paths ..."
+    tentacle configure --instance "$instanceName" --app "$applicationsDirectory"
+
+    echo "Configuring communication type ..."
+    if [[ ! -z "$ServerCommsAddress" || ! -z "$ServerPort" ]]; then
+        tentacle configure --instance "$instanceName" --noListen "True"
+    else
+        tentacle configure --instance "$instanceName" --port $internalListeningPort --noListen "False"
+    fi
+
+    echo "Updating trust ..."
+    tentacle configure --instance "$instanceName" --reset-trust
+
+    echo "Creating certificate ..."
+    tentacle new-certificate --instance "$instanceName" --if-blank
+}
+
+function registerTentacle() {
+    echo "Registering with server ..."
+
+    local ARGS=()
+
+    ARGS+=('register-k8s-cluster')
+
+    if [[ ! -z "$TargetEnvironment" ]]; then
+        IFS=',' read -ra ENVIRONMENTS <<<"$TargetEnvironment"
+        for i in "${ENVIRONMENTS[@]}"; do
+            ARGS+=('--environment' "$i")
+        done
+    fi
```

Comment:
We do support multiple environments/tenants via a command separated list in the environment variable in the container.